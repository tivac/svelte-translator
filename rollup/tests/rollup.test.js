const { rollup } = require("rollup");

const svelte2 = require("../svelte2.js");
const svelte3 = require("../svelte3.js");

describe("svelte-translator rollup plugins", () => {
    it("should be possible to use both plugins at once", async () => {
        const bundle = await rollup({
            input : "svelte2and3.js",

            plugins : [
                require("rollup-plugin-hypothetical")({
                    cwd   : __dirname,
                    files : {
                        "./svelte2and3.js" : `
                            import component2 from "./specimens/component.html";
                            import component3 from "./specimens/component.svelte";

                            console.log(component2, component3);
                        `,
                    },

                    allowFallthrough : true,
                }),

                require("rollup-plugin-node-resolve")(),

                svelte2(),
                svelte3(),
            ],
        });

        const { output } = await bundle.generate({
            format : "es"
        });

        const [{ code }] = output;

        expect(code).toMatch("console.log(SvelteComponent, Component);");
    });
});
