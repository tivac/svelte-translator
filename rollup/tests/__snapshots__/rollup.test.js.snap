// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svelte-translator rollup plugins should capture all warnings 1`] = `
Array [
  Object {
    "message": "A11y: <img> element should have an alt attribute",
    "plugin": "svelte-translator-rollup-svelte2",
    "pluginCode": "a11y-missing-attribute",
  },
  Object {
    "message": "A11y: <img> element should have an alt attribute",
    "plugin": "svelte-translator-rollup-svelte3",
    "pluginCode": "a11y-missing-attribute",
  },
]
`;
